source/tutorial/gsfa/GSFA.rst

```diff
@@ -0,0 +1,414 @@
+quadratic or linear functions. Highly complex function spaces
+:math:`\mathcal{F}` should be avoided because they result in
+overfitting.
+
+--------------
+
+.. _Linear GSFA Algorithm:
+
+
+5. Linear GSFA algorithm 
+------------------------
+
+In this section we will consider soulution of GSFA problem in the linear
+function space. Hence, the output components take the form
+
+.. math:: y_j(n) = \textbf{w}_j^{T} (\textbf{x}(n) - \hat{\textbf{x}})
+
+\ where
+
+.. math:: \hat{\textbf{x}} = \frac{1}{Q} \sum_n v_n \textbf{x}_n  \hspace{1cm} \text{(weighted average of all samples)}
+
+Thus, in the linear case, the SFA problem reduces to finding an optimal
+set of weight vectors :math:`w_j` under the constraints above, and it
+can be solved by linear algebra methods.
+
+As previously, suppose
+
+1\. Verticies
+   :math:`\textbf{V} = \{ \textbf{x}(1), \dots, \textbf{x}(N)\}` are the
+   input samples with weights :math:`\{v_1, \dots, v_N\}`, and
+
+2\. Edges :math:`\textbf{E}` be the set of edges
+   :math:`(\textbf{x}(n), \textbf{x}(n'))` with edge weights
+   :math:`\gamma_{n, n'}`. Also we imply that non-existing edges
+   :math:`(\textbf{x}(n), \textbf{x}(n')) \notin \textbf{E}` have
+   weights :math:`\gamma_{n, n'} = 0`
+
+
+
+**Step 1: Calculate covariance and second-moment matrices**
+
+
+The sample covariance matrix :math:`\textbf{C}_{G}` is defined as:
+
+.. math:: \textbf{C}_{G} := \frac{1}{Q} \sum_{n} v_n (\textbf{x}(n) - \hat{\textbf{x}})(\textbf{x}(n) - \hat{\textbf{x}})^T = \frac{1}{Q} \sum_{n} (v_n \textbf{x}(n) (\textbf{x}(n))^T ) - \hat{\textbf{x}} \hat{\textbf{x}}^T
+
+The derivative second-moment matrix :math:`\dot{\textbf{C}}_{G}` is
+defined as:
+
+.. math:: \dot{\textbf{C}}_{G} := \frac{1}{R} \sum_{n, n'} \gamma_{n, n'} (\textbf{x}(n') - \textbf{x}(n))(\textbf{x}(n') - \textbf{x}(n))^T
+
+
+**Step 2: Calculate sphering and rotation matricies**
+
+
+A shering matrix :math:`\textbf{S}` is computed with
+:math:`\textbf{S}^T \textbf{C}_{G} \textbf{S} = \textbf{I}`, then we
+derive that a sphered signal
+:math:`\textbf{z} := \textbf{S}^T \textbf{x}`.
+
+Afterwards, the :math:`J` directions of least variance in the derivative signal
+:math:`\dot{\textbf{z}}` are found and represented by an
+:math:`I \times J` rotation matrix :math:`\textbf{R}`, such that
+:math:`\textbf{R}^T \dot{\textbf{C}}_{z} \textbf{R} = \Lambda`, where
+:math:`\dot{\textbf{C}}_{z} := <\dot{\textbf{z}} \dot{\textbf{z}}^T>`
+and :math:`\Lambda` is a diagonal matrix with diagonal elements
+:math:`\lambda_1 \leq \lambda_2 \leq \dots \leq \lambda_J`.
+
+
+**Step 3: Calculate the weight matrix**
+
+
+Finally the algorithm returns the weight matrix
+:math:`W = (w_1, \dots, w_J)`, defined as
+
+.. math:: W = SR
+
+The features extracted:
+
+| 
+
+  .. math:: y = W^T (\textbf{x}(n) -  \hat{\textbf{x}}) \hspace{0.5cm}
+| where
+  :math:`\hspace{0.5cm} \Delta(y_j) = \lambda_j \hspace{0.5cm} 1 \leq j \leq J`
+
+--------------
+
+
+Referencies 
```

Comment:
Please check out the [section of footnotes and citations in the Sphinx-Docs](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#footnotes).